h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -15,12 +15,25 @@
 import java.io.IOException;
 import java.lang.reflect.Method;
 import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
 
 /**
  *  Model builder parent class.  Contains the common interfaces and fields across all model builders.
  */
 abstract public class ModelBuilder<M extends Model<M,P,O>, P extends Model.Parameters, O extends Model.Output> extends Iced {
 
+  Consumer<Model> _onSuccessCallback = null;
```

Comment:
What do you think about a single interface with 2 methods? IMHO that gives a little cleaner semantics. Is there a case where you would set just one of them?

Reply:
but then you would not be able to use lambdas!

Reply:
There is a way to satisfy both:

- Have an interface with two methods inside, with one setter.
- Introduce another "setter" method, accepting two lambdas, wrapping those in an anonymous class implementing the interface.

Sounds crazy even to me, but the price is low and people would be happy. I don't have a preference. Chose the way that was quickest for me to write.

Reply:
@Pscheidl this is my thinking:

- ModelBuilder is Iced, the callbacks are not declared `transient` which can lead to some very non-obvious issues because Consumer is not even Serializable by default
- IMHO it is easier to document a custom interface - the only place where to document in this code seems to be the setter methods which is not ideal

For both practical and semantics reasons I am leaning toward a custom interface / abstract class

Reply:
Used Iced abstract class as this is easier/quicker for any others to implement.